packs/duo/actions/ping.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from st2actions.runners.pythonrunner import Action
+import duo_client
+
+
+class Ping(Action):
+    def run(self):
+        """
+        Ping the Duo Platorm.
+
+        Returns: An dict with info returned by Duo.
+
+        Raises:
+          ValueError: On Auth Failure.
+        """
+
+        try:
+            ikey = self.config['ikey']
+            skey = self.config['skey']
+            host = self.config['host']
+        except KeyError:
+            raise ValueError("Duo config not found in config.")
+
+        auth = duo_client.Auth(ikey=ikey,
```

Comment:
Seems you're using this method over and over in each action, you should prob create a `lib` submodule and place it in there, look at how the other packs do it. When you add more functions it'll save you refactoring later.